jolt-core/src/poly/commitment/dory/commitment_scheme.rs

```diff
@@ -0,0 +1,345 @@
+//! Dory polynomial commitment scheme implementation
+
+use super::dory_globals::DoryGlobals;
+use super::jolt_dory_routines::{JoltG1Routines, JoltG2Routines};
+use super::wrappers::{
+    jolt_to_ark, ArkDoryProof, ArkFr, ArkG1, ArkGT, ArkworksProverSetup, ArkworksVerifierSetup,
+    JoltToDoryTranscript, BN254,
+};
+use crate::{
+    field::JoltField,
+    poly::commitment::commitment_scheme::{CommitmentScheme, StreamingCommitmentScheme},
+    poly::multilinear_polynomial::MultilinearPolynomial,
+    transcripts::Transcript,
+    utils::{errors::ProofVerifyError, math::Math, small_scalar::SmallScalar},
+};
+use ark_bn254::{G1Affine, G1Projective};
+use ark_ec::CurveGroup;
+use ark_ff::Zero;
+use dory::primitives::{
+    arithmetic::{Group, PairingCurve},
+    poly::Polynomial,
+};
+use rand_core::OsRng;
+use rayon::prelude::*;
+use std::borrow::Borrow;
+use tracing::trace_span;
+
+#[derive(Clone)]
+pub struct DoryCommitmentScheme;
+
+impl CommitmentScheme for DoryCommitmentScheme {
+    type Field = ark_bn254::Fr;
+    type ProverSetup = ArkworksProverSetup;
+    type VerifierSetup = ArkworksVerifierSetup;
+    type Commitment = ArkGT;
+    type Proof = ArkDoryProof;
+    type BatchedProof = Vec<ArkDoryProof>;
+    type OpeningProofHint = Vec<ArkG1>;
+
+    fn setup_prover(max_num_vars: usize) -> Self::ProverSetup {
+        let _span = trace_span!("DoryCommitmentScheme::setup_prover").entered();
+        let setup = ArkworksProverSetup::new_from_urs(&mut OsRng, max_num_vars);
+
+        // Initialize the prepared point cache for faster multi-pairings
+        // Skips cache initialization during tests to avoid shared state issues
+        #[cfg(not(test))]
```

Comment:
same thing here, I'd like to test this codepath. What are the shared state issues? Can they be fixed using [#[serial]](https://docs.rs/serial_test/latest/serial_test/)?